605CanPlaceFlowers.py

```
@@ -0,0 +1,57 @@
+    def canPlaceFlowers(self, flowerbed: list[int], n: int) -> bool:
+        if len(flowerbed)==1 and n==1:
+            return flowerbed[0]==0
+            
+        length=len(flowerbed)
+        for i in range(length-1):
+            if i==0 and flowerbed[i]==0 and flowerbed[i+1]==0 and n:
+                flowerbed[i]=1
+                n-=1
+            elif i==length-2 and flowerbed[i]==0 and flowerbed[i+1]==0 and n:
+                 flowerbed[i]=1
+                 n-=1
```

Ruff (0.12.2)

Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)